pkg/tasks/thanos_ruler.go

```diff
@@ -0,0 +1,185 @@
+		return errors.Wrap(err, "creating ThanosRuler Route failed")
+	}
+
+	_, err = t.client.WaitForRouteReady(r)
+	if err != nil {
+		return errors.Wrap(err, "waiting for ThanosRuler Route to become ready failed")
+	}
+
+	cr, err := t.factory.ThanosRulerClusterRole()
+	if err != nil {
+		return errors.Wrap(err, "initializing ThanosRuler ClusterRole failed")
+	}
+
+	err = t.client.CreateOrUpdateClusterRole(cr)
+	if err != nil {
+		return errors.Wrap(err, "reconciling ThanosRuler ClusterRole failed")
+	}
+
+	crb, err := t.factory.ThanosRulerClusterRoleBinding()
+	if err != nil {
+		return errors.Wrap(err, "initializing ThanosRuler ClusterRoleBinding failed")
+	}
+
+	err = t.client.CreateOrUpdateClusterRoleBinding(crb)
+	if err != nil {
+		return errors.Wrap(err, "reconciling ThanosRuler ClusterRoleBinding failed")
+	}
+
+	sa, err := t.factory.ThanosRulerServiceAccount()
+	if err != nil {
+		return errors.Wrap(err, "initializing ThanosRuler ServiceAccount failed")
+	}
+
+	err = t.client.CreateOrUpdateServiceAccount(sa)
+	if err != nil {
+		return errors.Wrap(err, "reconciling ThanosRuler ServiceAccount failed")
+	}
+
+	s, err := t.factory.ThanosRulerOauthCookieSecret()
+	if err != nil {
+		return errors.Wrap(err, "initializing ThanosRuler OAuth Cookie Secret failed")
+	}
+
+	err = t.client.CreateIfNotExistSecret(s)
+	if err != nil {
+		return errors.Wrap(err, "creating ThanosRuler OAuth Cookie Secret failed")
+	}
+
+	c := t.client.KubernetesInterface()
+	cm, err := c.CoreV1().Secrets(t.client.Namespace()).Get("grafana-datasources", metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to retrieve Grafana datasources config")
+	}
+	d := &manifests.GrafanaDatasources{}
+	err = json.Unmarshal(cm.Data["prometheus.yaml"], d)
+
+	hs, err := t.factory.ThanosRulerHtpasswdSecret(d.Datasources[0].BasicAuthPassword)
+	if err != nil {
+		return errors.Wrap(err, "initializing Thanos Ruler htpasswd Secret failed")
+	}
+
+	err = t.client.CreateIfNotExistSecret(hs)
+	if err != nil {
+		return errors.Wrap(err, "creating Thanos Ruler htpasswd Secret failed")
+	}
+
+	rs, err := t.factory.ThanosRulerRBACProxySecret()
+	if err != nil {
+		return errors.Wrap(err, "initializing Thanos Ruler RBAC proxy Secret failed")
+	}
+
+	err = t.client.CreateIfNotExistSecret(rs)
+	if err != nil {
+		return errors.Wrap(err, "creating Thanos Ruler RBAC proxy Secret failed")
+	}
+
+	{
+		// Create trusted CA bundle ConfigMap.
+		trustedCA, err := t.factory.ThanosRulerTrustedCABundle()
+		if err != nil {
+			return errors.Wrap(err, "initializing Thanos Ruler trusted CA bundle ConfigMap failed")
+		}
+
+		trustedCA, err = t.client.CreateIfNotExistConfigMap(trustedCA)
+		if err != nil {
+			return errors.Wrap(err, "creating Thanos Ruler trusted CA bundle ConfigMap failed")
+		}
+
+		// In the case when there is no data but the ConfigMap is there, we just continue.
+		// We will catch this on the next loop.
+		trustedCA = t.factory.HashTrustedCA(trustedCA, "thanos-ruler")
```

Comment:
cc @paulfantom i guess this covers the trusted ca bundle stuff

Reply:
I believe all the ca bundle stuff is there and working, but probably still needs some tweaks.